python/sdk/merlin/model.py

```diff
@@ -1128,20 +1128,27 @@ def create_prediction_job(self, job_config: PredictionJobConfig, sync: bool = Tr
         bar = pyprind.ProgBar(100, track_time=True,
                               title=f"Running prediction job {j.id} from model {self.model.name} version {self.id} "
                                     f"under project {self.model.project.name}")
+        retry = 5
         while j.status == "pending" or \
                 j.status == "running" or \
-                j.status == "terminating":
+                j.status == "terminating" or \
+                retry > 0:
             if not sync:
                 j = job_client.models_model_id_versions_version_id_jobs_job_id_get(model_id=self.model.id,
                                                                                    version_id=self.id,
                                                                                    job_id=j.id)
                 return PredictionJob(j, self._api_client)
             else:
-                j = job_client.models_model_id_versions_version_id_jobs_job_id_get(model_id=self.model.id,
-                                                                                   version_id=self.id,
-                                                                                   job_id=j.id)
+                try:
+                  j = job_client.models_model_id_versions_version_id_jobs_job_id_get(model_id=self.model.id,
+                                                                                       version_id=self.id,
```

Comment:
You should reset `retry` to 5 when the request is successful

Reply:
Done

Reply:
Done